### DISCUSSION TITLE
@tailwind utilities slows gatsby and nextjs dev builds to ~5 seconds

### DISCUSSION THREAD

**Discussion by ajsharp** [tailwindlabs/tailwindcss#2184] [Help]
I'm new to tailwind and have been experimenting with it in gatsby and nextjs. I'm including tailwind via postcss using the `@tailwind` directives. I've tested this on tailwind 1.6.3 and 1.7.0 and the performance observed is the same.

When I add all three tailwind directives to a css file, incremental development builds in either framework (saving a CSS file in development and waiting for it to re-compile the bundle) goes from taking ~500ms close to 5 seconds.

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

However, if I remove `@tailwind utilities`, the build times go back to roughly ~500ms.

I'm just using the base gatsby and nextjs installs/configs and running the development commands (`gatsby develop` / `yarn dev` in next). Using nextjs version 9.5.2 and gatsby version 2.24.41.

Has anyone else observed this behavior?

**Comment by ajsharp** [comment]
For now I'm including the production build in development via `@import` and incremental builds are very fast.

**Comment by AndyOGo** [comment]
We have the same problem.
Tailwind does not seem to support incremental builds, instead the whole thing is rebuild every time.

**Comment by simonswiss** [comment]
This could be related to [this PR/discussion happening](https://github.com/tailwindlabs/tailwindcss/pull/3032) - worth keeping an eye on that 👍
